contracts/erc721/PaymentObligation.sol

```diff
@@ -65,93 +60,41 @@ contract PaymentObligation is UserMintableERC721 {
    * @param _anchorId bytes32 The ID of the document as identified
    * by the set up anchorRegistry.
    * @param _merkleRoot bytes32 The root hash of the merkle proof/doc
-   * @param _collaboratorField string The values of the collaborator leaf
    * It needs to start with a collaborator prefix, ex: collaborator[0]
-   * @param _values string[5] The values of the leafs that is being proved
-   * precise-proofs library.
-   * @param _salts bytes32[5] The salts for the field that is being proved
+   * @param _values string[] The values of the leafs that are being proved
+   * using precise-proofs
+   * @param _salts bytes32[] The salts for the field that is being proved
    * Will be concatenated for proof verification as outlined in
    * precise-proofs library.
-   * @param _proofs bytes32[][5] Documents proofs that are needed
+   * @param _proofs bytes32[][] Documents proofs that are needed
    * for proof verification as outlined in precise-proofs library.
+   * @param _signature signed message used for on chain ACL
    */
   function mint(
```

Comment:
Can we remove `PaymentObligation.sol`? In the current implementation, it is just a wrapper for ERC721.
It stores the PODetails but they are stored in ERC721 transactions as well. 

What do you think? I might be cheaper.

Reply:
It contains the PO logic, the fields, what ever to decide to do with our identities and also getting information about the nft. PODetails are not stored in the ERCBase.

Reply:
Let say I only would use ERC721. 
I could pass the support_fields directly inside the constructor. 

To get the PODetails I could iterate over all transactions for the ERC721 address and looking for correct tokenId in the input.
(Alternative: We could implement it by storing it inside the events)